CHANGES.md

```diff
@@ -2,6 +2,434 @@
+* fixed a bug in BlockCyclic array accesses for 3D+ arrays
+* fixed a bug for membership queries in sparse CSR layouts
+* fixed a bug when storing max(int) in an associative domain of ints
+* fixed a bug in which arrays within records were incorrectly being localized
+* fixed a bug in which (+/-1)**k was incorrect for negative values of k
+* fixed a bug in advancing I/O channels
+* fixed bugs in the mapping of GMP functions down to the underlying C library
+* fixed a bug in which readline()'s 'amount' argument was used incorrectly
+* fixed a bug in which writeln(str) and writef("%s", str) behaved differently
+* fixed a bug related to using an 'enum' within a library module
+* fixed a bug ignoring whitespace when casting strings to values
+* fixed a bug in which the back-end compiler would warn about dead code
+* fixed a bug in which 'printchplenv --sh' was missing some variables
+* reduced cases when standard modules were automatically 'use'd in user code
+* fixed some issues with copy propagation
+* fixed an occasional compile-time failure in inserting wide references
+* fixed a bug in which sleep() could under-sleep by 1 microsecond
+* fixed some bugs with the --llvm-wide-opt flag
+* fixed a race condition in the 'pbs-aprun' launcher
+* improved how we handled environment variables with backtics in 'amudprun'
+* fixed a bug related to locale models and order of resolution
+
+Cray-specific Bug Fixes
+-----------------------
+TODO
+
+Third-Party Software Changes
+----------------------------
+* added 'libunwind' as a new third-party package in support of backtraces
+* upgraded jemalloc to version 4.2.1
+* upgraded GASNet to version 1.26.4
+* upgraded qthreads to version 1.11
+* upgraded hwloc to version 1.11.4
+* updated massivethreads to a new version
+* enabled decay-based purging in jemalloc
+* improved 'jemalloc's detection of 'librt' when using cray-prgenv-* compilers
+* disabled 'jemalloc's statistics gathering features by default
+* added support for the new Qthreads 'distrib' scheduler
+
+Launchers
+---------
+* added an --spmd flag to the 'mpirun' launcher for SPMD Chapel+MPI runs
+  (see TODO)
+* added --walltime, --partition, --exclude flags to 'slurm_gasnet_ibv' launcher
+
+Runtime Library Changes
+-----------------------
+* added the ability to implement atomics using C atomics ('cstdlib')
+  (see http://chapel.cray.com/docs/master/technotes/atomics.html and
+   http://chapel.cray.com/docs/1.14/usingchapel/chplenv.html#chpl-atomics)
+* re-enabled support for massivethreads for single-locale executions
+* added support for out-of-segment non-blocking puts and gets
+* changed the 'fifo' tasking layer to allocate task stacks in heap memory
+
+Generated Code
+--------------
+* improved generated code readability when using the new --denormalize flag
+* added function IDs and a function ID -> name table to the generated code
+* improved debug information for the LLVM back-end compiler
```

Comment:
--llvm couldn't make debug information before:

 -> enable the LLVM back-end compiler to produce debug information